api/event.js

```diff
@@ -0,0 +1,88 @@
+const fetch = require('node-fetch')
+const gqlr = require('graphql-request')
+const { GraphQLClient } = gqlr
+const { API_URL } = require('../src/config')
+
+const EndPoint = `${API_URL}/graphql`
+const GetParty = `
+  query getParty($address: String!) {
+    party(address: $address) {
+      id
+      name
+      description
+      headerImg
+      address
+    }
+  }
+`
+
+function htmlEncode(str) {
+  return str
+    ? str
+        .replace(/\&/g, '&amp;')
+        .replace(/\>/g, '&gt;')
+        .replace(/\</g, '&lt;')
+        .replace(/\"/g, '&quot;')
+    : ''
+}
+
+module.exports = async (req, res) => {
+  const paths = req.url.split('/')
+  const address = paths[paths.length - 1]
+
+  // fetch event info via graphql
+  const client = new GraphQLClient(EndPoint, {
+    headers: {
+      Authorization: ``
+    }
+  })
+  const { party } = await client.request(GetParty, { address })
+
+  // fetch index.html and replace placeholders
+  const baseUrl = `https://${req.headers.host}`
+  const r = await fetch(`${baseUrl}/build/index.html`)
```

Comment:
Can we run these in parallel as they don't have any dependency to each other. This will reduce our load times as they loading of the party meta and the downloading of our app can happen at the same time

Reply:
Good suggestions. I have added the parallel async call.